ui/src/main/webapp/app/query/query.html

```diff
@@ -13,17 +13,25 @@
   ~ See the License for the specific language governing permissions and
   ~ limitations under the License.
   -->
-<div layout-padding layout-margin ng-cloak>
-    <form name="ctrl.queryForm">
-        <operation-selector></operation-selector>
-        <seed-manager ng-show="ctrl.getSelectedOp().input"></seed-manager>
-        <date-range ng-if="ctrl.timeConfig && ctrl.getSelectedOp().view" conf="ctrl.timeConfig"></date-range>
-        <parameter-form ng-if="ctrl.getSelectedOp().parameters" parameters="ctrl.getSelectedOp().parameters"></parameter-form>
-        <view-builder ng-show="ctrl.getSelectedOp().view"></view-builder>
-        <directionality-picker ng-show="ctrl.getSelectedOp().inOutFlag"></directionality-picker>
-        <options-form ng-show="ctrl.hasOpOptions()"></options-form>
-    </form>
+<div layout="row" layout-padding flex>
+    <md-card flex>
+        <form name="ctrl.queryForm">
+            <div layout="row">
+                <div layout="column" flex="45"> <!-- left column -->
+                    <operation-selector></operation-selector>
+                    <seed-manager ng-show="ctrl.getSelectedOp().input"></seed-manager>
+                </div>
+                <div layout="column" flex="55"> <!-- right column -->
+                    <date-range ng-if="ctrl.timeConfig && ctrl.getSelectedOp().view" conf="ctrl.timeConfig"></date-range>
+                    <parameter-form ng-if="ctrl.getSelectedOp().parameters" parameters="ctrl.getSelectedOp().parameters"></parameter-form>
+                    <view-builder ng-show="ctrl.getSelectedOp().view"></view-builder>
+                    <directionality-picker ng-show="ctrl.getSelectedOp().inOutFlag"></directionality-picker>
+                    <options-form ng-show="ctrl.hasOpOptions()"></options-form>
+                </div>
+            </div>
+        </form>
+        <md-card-actions layout="row" layout-align="end center">
+            <md-button ng-click="ctrl.execute()" id="Execute Query" class="md-primary">Execute</md-button>
```

Comment:
This button is now docked at the bottom of the page. If your browser height is still not quite big enough you end up with just missing this button off and users may not realise they need to scroll down. Instead they might click the 'Execute all' button at the top - this just deletes their current query and reruns any previous queries.